errors/max-custom-routes-reached.md

```diff
@@ -0,0 +1,17 @@
+# Max Custom Routes Reached
+
+#### Why This Error Occurred
+
+The number of combined routes from `headers`, `redirects`, and `rewrites` exceeds 1024. This can impact the performance of requests since for each request all of these routes can potentially need to be checked.
```

Comment:
```suggestion
The number of combined routes from `headers`, `redirects`, and `rewrites` exceeds 1024. This can impact performance because each request will iterate over all routes to check for a match in the worst case.
```